### PR TITLE
Add foreign key relationships to moped_proj_phases

### PR DESCRIPTION
**Pull request by chiaberry**
## Associated issues
https://github.com/cityofaustin/atd-data-tech/issues/9242

Similar to moped_proj_milestones: https://github.com/cityofaustin/atd-moped/pull/654

I was unable to drop the phase_name column completely, because the project_list_view depends on it. I opened a new issue to address that tech debt: https://github.com/cityofaustin/atd-data-tech/issues/9350

I updated the seed data to have a phase_id in our seed projects. 

## Testing
**URL to test:**  test locally or moped test

**Steps to test:**
1. Add new phase in milestone table
2. Design and Construction have subphases, try adding one too. 
3. experiment with setting a current phase. Toggle another phase as current phase, only one phase can be current at any time. (But there can also be no current phases)
4. If the phase you set also matches one of our statuses, you should see the badge update to match the phase name
5. If you remove all current phases, you will see the badge label switch to active. 

---
#### Ship list
- [x] Code reviewed 
- [ ] Product manager approved
- [ ] Added to [QA test script](https://docs.google.com/spreadsheets/d/1n_O6MLh9cwwPf57HUM394Ea-z9uuoEV1-QW4axNZXLE/edit#) if applicable

### CODE REVIEW

File: moped-database/migrations/1653418384000_alter_table_moped_proj_phases/down.sql
Comment (chiaberry) [inline moped-database/migrations/1653418384000_alter_table_moped_proj_phases/down.sql:4]: https://austininnovation.slack.com/archives/CNUEPKLB1/p1653501387656529

File: moped-database/migrations/1653418384000_alter_table_moped_proj_phases/up.sql
Comment (frankhereford) [inline moped-database/migrations/1653418384000_alter_table_moped_proj_phases/up.sql:25]: Comments on columns! Thanks for leaving this hint for developers about what this is and isn't -- this is very helpful!

File: moped-database/migrations/1653418384000_alter_table_moped_proj_phases/up.sql
Comment (frankhereford) [inline moped-database/migrations/1653418384000_alter_table_moped_proj_phases/up.sql:14]: If there isn't a "current" column, this value gets set to _null_?

File: moped-editor/src/views/projects/projectView/ProjectTimeline.js
Comment (frankhereford) [inline moped-editor/src/views/projects/projectView/ProjectTimeline.js:258]: So much cleaner! 🧹

File: moped-editor/src/views/projects/projectView/ProjectTimeline.js
Comment (frankhereford) [inline moped-editor/src/views/projects/projectView/ProjectTimeline.js:165]: This comment makes me think that we're about to see a mutation being fired off one for each phase except for the "current" one. We'll never have enough phases for this to really be a big deal, but conceptually, ajax request in loops isn't ideal. Also, I totally recognize that this may be way outside the scope of your PR.


I feel like since only one phase is ever allowed to be the "current" one, the actual phase that is current is an intrinsic property of the project, not of the individual phases. Above, I see you're adding a field to track which phase is current in the project table, but I'm confused about what is going on here in this loop.

**Review by frankhereford — COMMENTED** [review]
Thank you for this @chiaberry! It looks awesome. 

I have a question below about how the current phase is being tracked in the database. Very briefly, I see the table `moped_projects` getting a integer column called `current_phase_id`, which feels very right to me, but I also see your comment on line 165 of `moped-editor/src/views/projects/projectView/ProjectTimeline.js` which implies to me that it's still tracked in the table of phase records. 

Maybe we can sync for a few minutes tomorrow morning and discuss what is going on? I am sure your explanation will make it all click into place for me in my head. 

I also observed a field that may need it's name changed?

I'm going to make this a "comment" review, and I'll submit an "approval" review as soon as I get my head around what's up with "current" tracking. Thank you so much for this work - you're paying back tech debt left and right!! 💸 🤑 🏦

File: moped-editor/src/views/projects/projectView/ProjectTimeline.js
Comment (mateoclarke) [inline moped-editor/src/views/projects/projectView/ProjectTimeline.js:117]: so much cleaner ✨

File: moped-editor/src/views/projects/projectView/ProjectTimeline.js
Comment (mateoclarke) [inline moped-editor/src/views/projects/projectView/ProjectTimeline.js:183]: 🙏

File: moped-editor/src/views/projects/projectView/ProjectTimeline.js
Comment (mateoclarke) [reply moped-editor/src/views/projects/projectView/ProjectTimeline.js:165]: This is an interesting point. Seems like your suggestion would be a bigger schema refactor. Maybe worth documenting in a new issue as tech debt.

File: moped-editor/src/views/projects/projectView/ProjectTimeline.js
Comment (mateoclarke) [inline moped-editor/src/views/projects/projectView/ProjectTimeline.js:535]: moving this out into its own function is a great move. thank you!

**Review by mateoclarke — APPROVED** [review]
🚢 💨 ✨ 

Looks good to me. Tested in local and reviewed the code. I'll approve assuming the discussion with Frank doesn't find any show stoppers. 

As we discussed in dev sync, I don't think we need to wait for @amenity or @johnclary's approval since this is purely DB and business logic related tech debt and is not implementing any new UI functionality. Maybe worth adding some new lines in the QA script though. Either way, I think we can merge once any concerns from Frank are resolved and the PMs can still review for regressions if needed even if the code is already merged.

File: moped-editor/src/views/projects/signalProjectTable/SignalProjectTable.js
Comment (chiaberry) [reply moped-editor/src/views/projects/signalProjectTable/SignalProjectTable.js:137]: Thank you for catching this

File: moped-editor/src/views/projects/projectView/ProjectTimeline.js
Comment (chiaberry) [reply moped-editor/src/views/projects/projectView/ProjectTimeline.js:165]: I was only updating the comment description for this function; I didn't change any of the functionality that existed before. 

If the new phase (or edited phase) is marked as current in the UI, then this function loops through all the phases for the project and if one is already marked as "is_current_phase", it sets it to false.

**Review by frankhereford — APPROVED** [review]
I'd still like to catch up over storing the "current" designation in the project vs. the phase records when it's a good time for you, but there's super no reason for me to hold up you merging whenever you're ready. Thank you again for all this great work!

File: moped-editor/src/views/projects/projectView/ProjectTimeline.js
Comment (mddilley) [inline moped-editor/src/views/projects/projectView/ProjectTimeline.js:158]: Echoing Frank and Mateo's similar comments that these kinds of changes are so nice to see!

File: moped-editor/src/views/projects/projectView/ProjectTimeline.js
Comment (mddilley) [reply moped-editor/src/views/projects/projectView/ProjectTimeline.js:183]: +1 🎣

File: moped-editor/src/views/projects/projectView/ProjectTimeline.js
Comment (mddilley) [inline moped-editor/src/views/projects/projectView/ProjectTimeline.js:191]: ✨

**Review by mddilley — COMMENTED** [review]
This all looks great, and it is awesome to see more normalization in the DB and to immediately see how it pays off in the frontend code. 🙌 

I did see one potential issue with the phase added to new projects. In `NewProjectView.js` on line 142, new projects are initialized with `phase_name: "potential"` when I think that could be changed to `phase_id: 1`. I think that could explain a small bug where "Potential" phase name isn't showing as expected (since its phase_id is set to 0). You can check this out at https://moped-test.austinmobility.io/moped/projects/227?tab=timeline. I deployed this branch to test earlier.

Screen Shot 2022-06-03 at 7 35 00 AM

**Pull request merged**
